recipes/ipaddress/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
+
+find_package(ipaddress CONFIG REQUIRED)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ipaddress::ipaddress)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_14)
```

This is setting C++ 14, but the recipe indicates min to be C++ 11 (And the readme does too)
```suggestion
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```

To ensure we work for the min version :)

Unless there's a reason this is set to 14, any insight would be appreciated in that case!

---

Hi, Rubén! I almost understand what needs to be done, but I want to clarify just in case:

```cmake
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```

does not install compilation with C++11. It only checks that the compiler supports C++11 and higher. In reality, the copying will be with the default version (usually 14 or 17).

In order to configure compilation with standard C++11, need to do the following:

```cmake
set_target_properties(${PROJECT_NAME}
     PROPERTIES
         CXX_STANDARD 11
)
```

or

```cmake
set_target_properties(${PROJECT_NAME}
     PROPERTIES
         CXX_STANDARD 11
         CXX_STANDARD_REQUIRED ON
         CXX_EXTENSIONS OFF
)
```

In this case, cmake will pass the necessary flag during compilation (`-std=gnu++11` or `-std=c++11` etc).

Do I need to set a minimum standard or force the test to compile with C++11? In other words, should I use `target_compile_features` or `set_target_properties`?

---

@VladimirShaleev Thank you for your feedback. The reason behind is because in Conan 1.x , the settings.compiler.cppstd is not configured by default, so the test package will use what is configured by the compiler as standard version (e.g. GCC5 will fail). In Conan 2.x, the c++ standard always follows what is configured by settings.compiler.cppstd, so it would not be needed to use `target_compile_features` neither `set_target_properties` in that case (future, when dropping Conan 1.x). But answering your question: `target_compile_features` is more than fine because it will check if the compiler supports c++11 and accommodate in case needed.

---

@uilianries Thanks for your reply! I'll follow your advice and save target_compile_features with cxx_std_11